File: core/src/com/shadyaardvark/hex/layouts/HexagonLayout.java
package com.shadyaardvark.hex.layouts;

import static java.lang.Math.max;
import static java.lang.Math.min;

import java.util.HashMap;

import com.badlogic.gdx.math.Vector2;
import com.shadyaardvark.hex.Hexagon;
import com.shadyaardvark.hex.Layout;

public class HexagonLayout implements Layout {
    @Override
    public HashMap<Vector2, Hexagon> getMap(int width, int height) {
        HashMap<Vector2, Hexagon> map = new HashMap<>(1000, 0.75f);

        for (int q = -width; q <= width; q++) {
            int r1 = max(-width, -q - width);
            int r2 = min(width, -q + width);
            for (int r = r1; r <= r2; r++) {
                map.put(new Vector2(q, r), new Hexagon(q, r));
            }
        }

        return map;
    }
}

File: core/src/com/trashcob/dicewars/map/layouts/HexagonLayout.kt
package com.trashcob.dicewars.map.layouts

import com.trashcob.dicewars.math.geom.hexagon.Hexagon


class HexagonLayout : GameBoardLayout {
    override fun generateLayout(width: Int, height: Int): List<Hexagon> {
        val hexagons = ArrayList<Hexagon>()

        for (q in -width..width) {
            val r1 = Math.max(-width, -q - width)
            val r2 = Math.min(width, -q + width)
            (r1..r2).mapTo(hexagons) { Hexagon(q, it) }
        }

        return hexagons
    }
}